source/mir/ndslice/slice.d

```diff
@@ -910,35 +910,37 @@ public:
     +/
     auto lightScope()() scope return @property
     {
-        return Slice!(LightScopeOf!Iterator, N, kind)(_structure, .lightScope(_iterator));
+        return Slice!(LightScopeOf!Iterator, N, kind, Labels)(_structure, .lightScope(_iterator), _labels);
     }
 
     /// ditto
     auto lightScope()() scope const return @property
     {
-        return Slice!(LightConstOf!(LightScopeOf!Iterator), N, kind)(_structure, .lightScope(_iterator));
+        return Slice!(LightConstOf!(LightScopeOf!Iterator), N, kind, Labels)(_structure,
+            .lightScope(_iterator), _labels);
```

Comment:
`Labels` -> `staticMap!(LightConstOfLightScopeOf, Labels)`,
`_labels` -> `CallForEach!(.lightScope)(_labels)`,
where
`private alias LightConstOfLightScopeOf(Iterator) = LightConstOf!(LightScopeOf!Iterator)`
and
private `CallForEach` to be implemented the similar way to `forward` from `mir.functional`, mir-core package.

Reply:
Sorry @9il but i cannot figure out how to modify `forward` for this exact functionality.

Reply:
draft:
```d
template forward(args...)
{
    static if (args.length)
    {
        alias arg = args[0];
            @optmath @property fwd()()
            {
                    import mir.qualifier : lightScope;
                    return lightScope(arg);
            }
        alias forward = AliasSeq!(fwd, forward!(args[1..$]));
    }
    else
        alias forward = AliasSeq!();
}

```

Reply:
@optmath can be removed

Reply:
I understood it expands the arguments but I am lost while extending it's functionality with `.lightScope`

Reply:
thanks a lot!